dashboards-observability/public/components/visualizations/charts/bar/bar.tsx

```diff
@@ -4,80 +4,78 @@
-  const DEFAULT_LABEL_SIZE = 10;
   const { vis } = visualizations;
   const {
     data,
     metadata: { fields },
   } = visualizations.data.rawVizData;
   const lastIndex = fields.length - 1;
   const {
-    dataConfig = {},
+    dataConfig: {
+      chartStyles = {},
+      valueOptions = {},
+      legend = {},
+      colorTheme = [],
+      panelOptions = {},
+    },
     layoutConfig = {},
     availabilityConfig = {},
   } = visualizations?.data?.userConfigs;
-  const dataConfigTab =
-    visualizations.data?.rawVizData?.bar?.dataConfig &&
-    visualizations.data.rawVizData.bar.dataConfig;
-  const xaxis = dataConfig?.valueOptions?.dimensions
-    ? dataConfig.valueOptions.dimensions.filter((item) => item.label)
-    : [];
-  const yaxis = dataConfig?.valueOptions?.metrics
-    ? dataConfig.valueOptions.metrics.filter((item) => item.label)
-    : [];
-  const barOrientation = dataConfig?.chartStyles?.orientation || vis.orientation;
-  const isVertical = barOrientation === vis.orientation;
+  const xaxis = valueOptions.dimensions ? filterDataConfigParameter(valueOptions.dimensions) : [];
+  const yaxis = valueOptions.metrics ? filterDataConfigParameter(valueOptions.metrics) : [];
+  const isVertical = vis.orientation === 'v';
   let bars, valueSeries, valueForXSeries;
 
   if (!isEmpty(xaxis) && !isEmpty(yaxis)) {
-    valueSeries = isVertical ? [...yaxis] : [...xaxis];
-    valueForXSeries = isVertical ? [...xaxis] : [...yaxis];
+    valueSeries = [...yaxis];
+    valueForXSeries = [...xaxis];
   } else {
     return <EmptyPlaceholder icon={visualizations?.vis?.icontype} />;
   }
-
-  const tickAngle = dataConfig?.chartStyles?.rotateBarLabels || vis.labelangle;
-  const lineWidth = dataConfig?.chartStyles?.lineWidth || vis.linewidth;
+  const tickAngle = chartStyles.rotateBarLabels || vis.labelangle;
+  const lineWidth = chartStyles.lineWidth || vis.linewidth;
   const fillOpacity =
-    dataConfig?.chartStyles?.fillOpacity !== undefined
-      ? dataConfig?.chartStyles?.fillOpacity / FILLOPACITY_DIV_FACTOR
-      : vis.fillOpacity / FILLOPACITY_DIV_FACTOR;
-  const barWidth = 1 - (dataConfig?.chartStyles?.barWidth || vis.barwidth);
-  const groupWidth = 1 - (dataConfig?.chartStyles?.groupWidth || vis.groupwidth);
-  const showLegend = !(
-    dataConfig?.legend?.showLegend && dataConfig.legend.showLegend !== vis.showlegend
-  );
-  const legendPosition = dataConfig?.legend?.position || vis.legendposition;
-  visualizations.data?.rawVizData?.dataConfig?.metrics
-    ? visualizations.data?.rawVizData?.dataConfig?.metrics
-    : [];
-  const labelSize = dataConfig?.chartStyles?.labelSize || DEFAULT_LABEL_SIZE;
+    chartStyles.fillOpacity !== undefined
+      ? chartStyles.fillOpacity / FILLOPACITY_DIV_FACTOR
+      : vis.fillopacity / FILLOPACITY_DIV_FACTOR;
+  const barWidth = 1 - (chartStyles.barWidth || vis.barwidth);
+  const groupWidth = 1 - (chartStyles.groupWidth || vis.groupwidth);
```

Comment:
Be careful while using || operators to provide default values for numeric fields. The second value gets assigned in case the first value remains 0 (treated as falsy value). Although the user may want to set the value 0, that never gets assigned and the default value is chosen. Make sure, your code does not have the problem.